include/deal.II/lac/petsc_vector_base.h

```diff
@@ -848,8 +863,10 @@ namespace PETScWrappers
 
       const PetscInt petsc_i = index;
 
+      PetscScalar cast_value = static_cast<PetscScalar> (value);
+
       const int ierr
-        = VecSetValues (vector, 1, &petsc_i, &value, INSERT_VALUES);
+        = VecSetValues (vector, 1, &petsc_i, &cast_value, INSERT_VALUES);
```

Comment:
As a side remark, there is also `VecSetValues` that is meant for the addition of a single value and that could be used here: http://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Vec/VecSetValue.html#VecSetValue . But this is for a separate patch.